### PR TITLE
Add Travis CI config and fix Junit tests run from Gradle

### PR DESCRIPTION
**Pull request by DanVanAtta**
Merging this PR will allow Travis CI to automatically build the tripleA project whenever anyone pushes content to it. This PR will also provide a fix to unit tests run in gradle (unit tests still pass via old mechanisms, ant and eclipse)

Adding travis CI config is pretty trivial, only one required line (jdk lines are optional):
```
language: java
jdk:
  - oraclejdk8
  - oraclejdk7
```
Once Travis CI has been given permission on the github repo, it does everything from there. It picked up the travis CI config file with no problem. The build failed initially. The output from Travis was identical to local gradle. I then fixed gradle local, pushed, and travis picked it up and was green. I merged the travis file into another local branch, pushed the updated branch, and Travis CI started building that automatically within a few moments. Here is what it looked like when it picked up the updated branch:

travis_ci_build_history_example

For the tests...
Two test fixes required. One test searches what it thinks are only compiled files. Introduce a non-java compiled file into this folder tree and you get an error.

The vast majority of tests were  broken in gradle due to a classpath problem. We have some test XML files that used to live alongside unit test java files and were being loaded with `getResource("mapFileName.xml")`. Gradle was not picking them up. Maybe it would if we added the package as a path. I like moving the test data to a home away from the source code. Allows for better discovery of it, not so hidden.

[Review on Reviewable](https://reviewable.io/reviews/triplea-game/triplea/58)

### CODE REVIEW

**Comment by gaborbernat** [comment]
Absolutely supporting this :+1:  will fix the Gradle issues

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+language: java
+jdk:
+  - oraclejdk8
+  - oraclejdk7
```
Comment (veqryn) [inline .travis.yml]: java 6?

**Comment by veqryn** [comment]
i don't see any file changes that are different than 43...?

**Comment by DanVanAtta** [comment]
How do you mean by 43 @veqryn ? there are 2 changed files, one new one, and a dozen moved XMLs.

**Comment by DanVanAtta** [comment]
@veqryn  sorry, I assume you mean PR #43. #43 does have different stuff from this, it this change plus more. The branch for this PR is merged into that one. #43 is built on top of this PR. So if you merge this PR, #43 becomes simpler.

All straight code changes will be based on this branch from me, so you're going to see this diff everywhere until this branch (or one of the branches built on top of it) are merged. Same with #43, but #43 is used as a base branch for tests. So new tests you'll see the changes #58 and #43 as well as any that are in the additional branch.

Hence why #43 and #58 are good to get merged first, simplifies the other PRs that build on top.

**Comment by DanVanAtta** [comment]
Let me restate my prior comment:
- #58 is the base branch. #43 is #48 plus a little bit more for testing. From there all test updates are based the branch in #43. So if you look at test changes you'll see everything in #43. #58 is the base branch for non-test changes.

As you can see this is not necessarily simple. Bottom line is that if you merge this PR, all the other ones will "get simpler" and you'll see less in those Diffs. By reviewing all commits in #43, even those that are in here, you've effectively already reviewed this PR.

**Comment by DanVanAtta** [comment]
Review status: 0 of 49 files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

<sup>**[.travis.yml, line 4 \[r1\]](https://reviewable.io:443/reviews/triplea-game/triplea/58#-Jw95Zmc8dqE2-cirW5E-r1-4)** ([raw file](https://github.com/triplea-game/triplea/blob/aa26afaab73979a82dfac5740b9fd72b9c2d0b8a/.travis.yml#L4)):</sup>
Seems like my comment/response to this got lost. As far as JDK6 is concerned, only openjdk6 is available. Oraclejdk6 is no longer available, end of life. 

I do not think we have the resources to fix an openJDK6 only test failure. So the result of an openjdk6 build is meaningless, thus not included. (nor are the openjdk's for 7 and 8)

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/triplea-game/triplea/58)

**Comment by veqryn** [comment]
unless it is a bug with openjdk itself, i think we still need to support java 6.  and if we run into a bug with openjdk that is specific to openjdk and not manually replicate-able with oracle java 6, then we can ignore that specific bug if the fix for it is not easy.

**Pull request merged**